File: testbed/test/cruise/test/UnidirectionalOneTest.java
package cruise.test;

import org.junit.*;

public class UnidirectionalOneTest
{

  @Test
  public void ConstructorManySide()
  {
    StudentAH s = new StudentAH(1);
    MentorAH m = new MentorAH("a",s);
    Assert.assertEquals("a",m.getName());
    Assert.assertEquals(1, m.getStudent().getNumber());
  }
  
  @Test(expected=RuntimeException.class)
  public void ConstructorSetNull()
  {
    new MentorAH("a",(StudentAH)null);
  }   
  
  @Test
  public void deleteLeavesStudentAlone()
  {
    MentorAH m = new MentorAH("a",new StudentAH(1));
    StudentAH s = m.getStudent();
    ProgramAH p = new ProgramAH();
    s.setProgram(p);
    m.delete();
    Assert.assertEquals(null, m.getStudent());
    Assert.assertEquals(null, m.getProgram());
    Assert.assertEquals(p, s.getProgram());
    
  }
  
}

File: testbed_python/test/associations/unidirectional_one_test.py
import unittest

# Test classes are imported dynamically
# Must be generated into ImportModules' namespace then imported
from ImportModules import importModules

importModules(["MentorAH", "ProgramAH", "StudentAH"], ["associations"])
from ImportModules import *

class UnidirectionalOneTest(unittest.TestCase):
    def test_constructorManySide(self):
        s = StudentAH.StudentAH(1)
        m = MentorAH.MentorAH("a", s)
        self.assertEqual("a", m.getName())
        self.assertEqual(1, m.getStudent().getNumber())
    
    def test_constructorSetNull(self):
        with self.assertRaises(RuntimeError):
            MentorAH.MentorAH("a", None)
    
    def test_deleteLeavesStudentAlone(self):
      m = MentorAH.MentorAH("a", StudentAH.StudentAH(1))
      s = m.getStudent()
      p = ProgramAH.ProgramAH()
      s.setProgram(p)
      del m
      self.assertEqual(p, s.getProgram())
